src/code/z_actor.c

```diff
@@ -447,101 +449,128 @@ void Target_Draw(TargetContext* targetCtx, PlayState* play) {
-        f32 temp3 =
-            (unkActor->world.pos.y + (unkActor->targetArrowOffset * unkActor->scale.y)) - targetCtx->naviRefPos.y;
-        f32 temp4 = unkActor->world.pos.z - targetCtx->naviRefPos.z;
+    if (!Math_StepToF(&targetCtx->naviMoveProgressFactor, 0.0f, 0.25f)) {
+        f32 moveScale = 0.25f / targetCtx->naviMoveProgressFactor;
+        f32 x = actor->world.pos.x - targetCtx->naviHoverPos.x;
+        f32 y = (actor->world.pos.y + (actor->targetArrowOffset * actor->scale.y)) - targetCtx->naviHoverPos.y;
+        f32 z = actor->world.pos.z - targetCtx->naviHoverPos.z;
 
-        targetCtx->naviRefPos.x += temp2 * temp1;
-        targetCtx->naviRefPos.y += temp3 * temp1;
-        targetCtx->naviRefPos.z += temp4 * temp1;
+        targetCtx->naviHoverPos.x += x * moveScale;
+        targetCtx->naviHoverPos.y += y * moveScale;
+        targetCtx->naviHoverPos.z += z * moveScale;
     } else {
-        Target_SetNaviState(targetCtx, unkActor, actorCategory, play);
+        // Set Navi pos and color after reaching destination
+        Target_SetNaviState(targetCtx, actor, category, play);
     }
 
-    if ((actorArg != NULL) && (targetCtx->reticleSpinCounter == 0)) {
-        Actor_ProjectPos(play, &actorArg->focus.pos, &projectedFocusPos, &cappedInvWDest);
+    // Release lock-on if the actor is off screen.
+    // The camera is always moving toward the locked-on actor, so it seems difficult
+    // to move the actor off screen, if its even possible.
+    if ((curLockOnActor != NULL) && (targetCtx->reticleSpinCounter == 0)) {
+        Actor_ProjectPos(play, &curLockOnActor->focus.pos, &projectedFocusPos, &cappedInvWDest);
+
         if (((projectedFocusPos.z <= 0.0f) || (1.0f <= fabsf(projectedFocusPos.x * cappedInvWDest))) ||
             (1.0f <= fabsf(projectedFocusPos.y * cappedInvWDest))) {
-            actorArg = NULL;
+            curLockOnActor = NULL;
         }
     }
 
-    if (actorArg != NULL) {
-        if (actorArg != targetCtx->lockOnActor) {
+    if (curLockOnActor != NULL) {
+        if (curLockOnActor != targetCtx->reticleActor) {
             s32 lockOnSfxId;
 
-            Target_InitReticle(targetCtx, actorArg->category, play);
-            targetCtx->lockOnActor = actorArg;
+            // Set up a new lock-on
+            Target_InitReticle(targetCtx, curLockOnActor->category, play);
+            targetCtx->reticleActor = curLockOnActor;
 
-            if (actorArg->id == ACTOR_EN_BOOM) {
+            if (curLockOnActor->id == ACTOR_EN_BOOM) {
+                // Don't draw the reticle when locked onto the boomerang.
+                // Note that it isn't possible to lock onto the boomerang, so this code doesn't do anything.
+                // This implies that the boomerang camera lock may have been implemented with Z-Targeting at one point,
+                // but was eventually implemented as its own camera mode instead.
                 targetCtx->reticleFadeAlphaControl = 0;
             }
```

Comment:
For anyone curious on my claims here, I removed `CAM_MODE_FOLLOW_BOOMERANG` camera mode and made the boomerang manually targetable to make this code actually able to manifest:

https://github.com/user-attachments/assets/c36899c7-afd5-47b1-a0f0-c5fa2f784c3b